optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -512,7 +512,10 @@ def _tune_params(
         sampler: optuna.samplers.BaseSampler,
         step_name: str,
     ) -> _OptunaObjective:
-        pbar = tqdm.tqdm(total=n_trials, ascii=True)
+        if self.auto_options["verbosity"] == 0:
+            pbar = None
+        else:
+            pbar = tqdm.tqdm(total=n_trials, ascii=True)
```

Comment:
This "verbosity" decision is far away from the existing verbosity decision in `_LightGBMBaseTuner.run()` method (Line 436-441). How about moving this logic into the `if` statement in `_LightGBMBaseTuner.run()` method, and add the argument of `pbar` for `_tune_params` methods? My proposed suggestion looks as follows.

```python
...
# In `_LightGBMBaseTuner.run()`
if self.auto_options["verbosity"] == 0:
    optuna.logging.disable_default_handler()
    self.auto_options["pbar"] = None
    self.lgbm_params["verbose"] = -1
    self.lgbm_params["seed"] = 111
    self.lgbm_kwargs["verbose_eval"] = False
else:
     self.auto_options["pbar"] = tqdm.tqdm(total=n_trials, ascii=True)
...
# In `_LightGBMBaseTuner._tune_params()`
pbar = self.auto_options["pbar"]
...
```

Reply:
Thank you for your proposal. I think it improves the locality of `self.auto_options["verbosity"]`, but it loses the locality of `pbar`. Currently, the `auto_options` dictionary handles literals only, but it may increase the potential of issues if we mix the instance like `tqdm`. For example, I think we need to remove it before pickling the `LightGBMTuner`.
What do you think?

Reply:
Thanks for the reply. I agree with your idea not to use `auto_options` for this purpose. How about introducing another attribute `_use_pbar` for the locality of both `self.auto_options["verbosity"]` and `pbar` as follows?

```python
# In `_LightGBMBaseTuner.__init__()`
self._use_pbar = True
...
# In `_LightGBMBaseTuner.run()`
if self.auto_options["verbosity"] == 0:
    optuna.logging.disable_default_handler()
    self._use_pbar = False
    self.lgbm_params["verbose"] = -1
    self.lgbm_params["seed"] = 111
    self.lgbm_kwargs["verbose_eval"] = False
...
# In `_LightGBMBaseTuner._tune_params()`
pbar = tqdm.tqdm(total=n_trials, ascii=True) if self._use_pbar else None
```

Reply:
Thank you for your suggestion. That makes sense. I applied the change in https://github.com/optuna/optuna/pull/1460/commits/9895eafe88ca5629aca7e192c51cfb3f11867638.